### PR TITLE
Make packaged kedro project return `session.run()` output

### PR DESCRIPTION
**Pull request by ElenaKhaustova**
## Description
Solves https://github.com/kedro-org/kedro/issues/2681

Should be reviewed and merged with: https://github.com/kedro-org/kedro-starters/pull/235

## Development notes
Modified project run command to return `session.run()` output in the interactive environment

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

**Comment by ankatiyar** [comment]
I think you might need to make the same change in the blank project template as well - https://github.com/kedro-org/kedro/blob/main/kedro/templates/project/%7B%7B%20cookiecutter.repo_name%20%7D%7D/src/%7B%7B%20cookiecutter.python_package%20%7D%7D/__main__.py

**Review by ankatiyar — APPROVED** [review]
Tested locally! 👍🏾

File: RELEASE.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Upcoming Release
 
 ## Major features and improvements
+* Made packaged kedro project return `session.run()` output when running it in the interactive environment.
```
Comment (merelcht) [inline RELEASE.md:4]:
```suggestion
* Made packaged kedro project return `session.run()` output to be used when running it in the interactive environment.
```

File: docs/source/tutorial/package_a_project.md [hunk 2]
```diff
@@ -144,6 +144,19 @@ main(
 
 This is equivalent to `python -m <package_name>` at the command line, and you can pass in all the arguments that correspond to the options described by `python -m <package_name> --help`.
 
+```{note}
+If you run the packaged project in the interactive environment like ipython or Databricks you can also consume the output of the `main()`
```
Comment (merelcht) [inline docs/source/tutorial/package_a_project.md:148]:
```suggestion
If you run the packaged project in the interactive environment like IPython or Databricks you can also consume the output of the `main()`
```

File: docs/source/tutorial/package_a_project.md [hunk 3]
```diff
@@ -144,6 +144,19 @@ main(
...
+which returns the `session.run()` output.
+```
+
+```python
+from spaceflights.__main__ import main
+
+def run_kedro_pipeline():
+   result = main(pipeline_name=<pipeline>)
+   do_something_about(<result>)
```
Comment (merelcht) [inline docs/source/tutorial/package_a_project.md:157]:
```suggestion
   do_something_with(<result>)
```

**Review by merelcht — APPROVED** [review]
Tested and works as expected! 👍 Left some small suggestions.

File: docs/source/tutorial/package_a_project.md (hunk 2 above)
Comment (astrojuanlu) [inline docs/source/tutorial/package_a_project.md:148]:
```suggestion
If you run the packaged project in the interactive environment like IPython or Databricks you can also consume the output of the `main()`
```

File: RELEASE.md (hunk 1 above)
Comment (astrojuanlu) [inline RELEASE.md:4]:
```suggestion
* Made packaged Kedro projects return `session.run()` output when running it in the interactive environment.
```

**Review by astrojuanlu — APPROVED** [review]
Thanks @ElenaKhaustova! Added a couple of nitpicks to the docs and release notes

**Pull request merged**